### ISSUE TITLE
[Question] Still usable with binary images as masks?

### ISSUE THREAD

**Issue by Gloupys** [matterport/Mask_RCNN#316]
Hello!

I recently saw this implementation of the mask RCNN and wanted to try it out on my aerial imagery dataset. It is basically a whole lot of satellite images and the goal is to segment the buildings on it (later on, we want to be able to get the forests as well).

My datasets are composed of these images and binary images representing the masks (basically black background with the shape of the buildings in white) and I wanted to know if it could be used with this implementation. 

Thanks in advance!

**Comment by mnboos** [comment]
Yes, it is. It's exactly what I'm currently doing in my master thesis.

Have a look at my work in progress: https://github.com/mnboos/osm-instance-segmentation/blob/master/inspect_model.ipynb

One very tricky part is how to get plausible (i.e. mostly rectangular) outlines from the predicted masks: https://github.com/mnboos/osm-instance-segmentation/blob/master/Regularization.ipynb

For the generation of the training data, I wrote a little tool: https://github.com/mnboos/airtiler
